src/lib/cimbar_js/cimbar_zstd_js.cpp

```diff
@@ -0,0 +1,62 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#include "cimbar_zstd_js.h"
+
+#include "compression/zstd_decompressor.h"
+
+#include <memory>
+#include <sstream>
+
+// wasm api will be something like
+// zstd_init_stream(bytes) -> constructor, reads skippable frame at start? (needs to for ofstream case)
+// get_filename(buff, size) -> cached in wasmjs api from constructor call?
+// get_bytes(buff, size) -> "decompress_chunk()/write()"
+
+namespace {
+	// maybe a map eventually
+	std::unique_ptr<cimbar::zstd_decompressor<std::stringstream>> _dec;
```

Comment:
I'm also thinking this should be a threadlocal? Perhaps? It's obnoxious to hold onto the state in general -- this is a compromise between various design choices around mapping a synchronous api (this one) to the async javascript "ReadableStream" interface. There are other options that might result in a cleaner C api, but cleaner is not always better imo (in general, I'd rather the C++ code take on the burden of jumping through more hoops than the JS does, since C++ is a more expressive language)